classes/Combination.php

```diff
@@ -275,8 +275,14 @@ public function deleteAssociations()
         if ((int) $this->id === 0) {
             return false;
         }
-        $result = Db::getInstance()->delete('product_attribute_combination', '`id_product_attribute` = ' . (int) $this->id);
-        $result &= Db::getInstance()->delete('product_attribute_image', '`id_product_attribute` = ' . (int) $this->id);
+        $result = Db::getInstance()->delete(
+            'product_attribute_combination',
+            '`id_product_attribute` = ' . (int) $this->id
+            )
+            && Db::getInstance()->delete(
+                'product_attribute_image',
+                '`id_product_attribute` = ' . (int) $this->id
+            );
```

Comment:
This changes the behaviour If you perform both actions in the same logical `&&` and the first one fails then the second one won't be executed.
To keep the same behaviour you need to split this into two lines.
```suggestion
        $result = Db::getInstance()->delete('product_attribute_combination', '`id_product_attribute` = ' . (int) $this->id);
        $result = $result && Db::getInstance()->delete('product_attribute_image', '`id_product_attribute` = ' . (int) $this->id);
```

Reply:
Fixed.